index.js

```diff
@@ -31,10 +33,22 @@ module.exports = (fn, opts) => {
 
 		const ret = fn.apply(null, arguments);
 
-		cache.set(key, {
-			data: ret,
-			maxAge: Date.now() + (opts.maxAge || 0)
-		});
+		if (isPromise(ret) && opts.cacheRejection === false) {
+			// Only cache resolved promises unless `cacheRejection` is set to `true`
+			ret
+				.then(() => {
+					cache.set(key, {
+						data: ret,
+						maxAge: Date.now() + (opts.maxAge || 0)
+					});
+				})
+				.catch(() => { });
+		} else {
+			cache.set(key, {
```

Comment:
This is duplicated. Can you extract it into an arrow function above the `if` statement on line 36?